examples/gno.land/r/matijamarjanovic/home/home.gno

```diff
@@ -0,0 +1,230 @@
+knowledge to real-world projects and contribute to
+innovative technology solutions.
+
+In addition to my academic pursuits, I enjoy traveling and staying active through weightlifting. I have a keen interest in electronic music and often explore various genres. I believe in maintaining a balanced lifestyle that complements my professional development.`
+	redVotes = 0
+	greenVotes = 0
+	blueVotes = 0
+	currentColor = `rgb(120, 120, 120)`
+	blueLink = "https://www.google.com"
+	redLink = "https://www.google.com"
+	greenLink = "https://www.google.com"
+}
+
+func UpdatePFP(url, caption string) {
+	AssertAuthorized()
+	pfp = url
+	pfpCaption = caption
+}
+
+func UpdateAboutMe(col1 string) {
+	AssertAuthorized()
+	abtMe = col1
+}
+
+func maxOfThree(a, b, c int64) int64 {
+	max := a
+	if b > max {
+		max = b
+	}
+	if c > max {
+		max = c
+	}
+	return max
+}
+
+func calculateRGB(red, green, blue int64) (int, int, int) {
+	maxValue := maxOfThree(red, green, blue)
+
+	if maxValue > 255 {
+		scaleFactor := float64(255) / float64(maxValue)
+
+		red = int64(float64(red) * scaleFactor)
+		green = int64(float64(green) * scaleFactor)
+		blue = int64(float64(blue) * scaleFactor)
+	}
+
+	return int(red), int(green), int(blue)
+}
+
+func VoteRed() {
+
+	ugnotAmount := std.GetOrigSend().AmountOf("ugnot")
+
+	redVotes = redVotes + ugnotAmount/750
+	red, green, blue := calculateRGB(redVotes, greenVotes, blueVotes)
+	currentColor = ufmt.Sprintf(`rgb(%s, %s, %s)`, strconv.Itoa(red), strconv.Itoa(green), strconv.Itoa(blue))
+}
+
+func VoteGreen() {
+	ugnotAmount := std.GetOrigSend().AmountOf("ugnot")
+
+	greenVotes = greenVotes + ugnotAmount/750
+	red, green, blue := calculateRGB(redVotes, greenVotes, blueVotes)
+	currentColor = ufmt.Sprintf(`rgb(%s, %s, %s)`, strconv.Itoa(red), strconv.Itoa(green), strconv.Itoa(blue))
+}
+
+func VoteBlue() {
+	ugnotAmount := std.GetOrigSend().AmountOf("ugnot")
+
+	blueVotes = blueVotes + ugnotAmount/750
+	red, green, blue := calculateRGB(redVotes, greenVotes, blueVotes)
+	currentColor = ufmt.Sprintf(`rgb(%s, %s, %s)`, strconv.Itoa(red), strconv.Itoa(green), strconv.Itoa(blue))
+
+}
+
+func CollectBalance() {
+	AssertAuthorized()
+
+	banker := std.GetBanker(std.BankerTypeRealmSend)
+	ownerAddr := Address()
+
+	banker.SendCoins(std.CurrentRealm().Addr(), ownerAddr, banker.GetCoins(std.CurrentRealm().Addr()))
+}
+
+func Render(path string) string {
+
+	out := ufmt.Sprintf("<div style='font-family: \"Trebuchet MS\", sans-serif;font-size: 5rem; color: %s;'>\n\n", currentColor)
+	out += ufmt.Sprintf("# Matija's Homepage \n\n%s", "")
+	out += "</div>\n\n"
+	out += "<div style='font-family: \"Trebuchet MS\", sans-serif; text-align: justify;'>\n\n"
+	out += renderAboutMe()
+	out += "\n\n"
+	out += ufmt.Sprintf("<h3 style='color: %s;'>Vote for the color of the page - the more you pay the more effect it will have on the homepage ;)</h3>\n", currentColor)
+	out += renderButtons()
+	out += "</div>"
+	out += renderGitHubProjects()
+	out += "\n\n"
```

Comment:
Does gno have templates? This could be much cleaner.

Also you have a Sprintf with an empty string there in the second line.

Reply:
To my knowledge - (g)no.
Thanks for pointing out, I must have missed the empty string. Fixed it in one of the newer commits.